### PR TITLE
Validate network configuration after installation

### PR DESCRIPTION
**Pull request by jknphy**
See [poo#88313](https://progress.opensuse.org/issues/88313)

[Verification run](https://openqa.suse.de/tests/overview?build=jknphy%2Fos-autoinst-distri-opensuse%23validate_network_configuration&distri=sle&version=15-SP3)

### CODE REVIEW

File: tests/console/yast2_kdump.pm [hunk 1]
```diff
@@ -36,7 +36,7 @@ sub run {
     # check service (without restarting)
     systemctl('is-enabled kdump');
 
-    validate_cfg_file(get_test_suite_data()->{configuration_files});
+    validate_cfg_file(get_test_suite_data()->{config_files});
```
Comment (rwx788) [inline tests/console/yast2_kdump.pm:39]: Could you please split this change to the separate commit?

File: test_data/yast/installation/network_config_s390x_svirt.yaml [hunk 2]
```diff
@@ -0,0 +1,30 @@
+---
+network:
+  network_manager: wicked
+  device: eth0
+  config_files:
+    - path: /etc/sysconfig/network/ifcfg-eth0
+      entries:
+        - BOOTPROTO='dhcp'
```
Comment (rwx788) [inline test_data/yast/installation/network_config_s390x_svirt.yaml:8]: Hmm, I wonder if we need to keep the key-pair approach we use in kdump or just use the way to check complete lines? As here we could have used mapping. WDYT?

File: tests/console/verify_network.pm [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+        qw(network_manager device config_files)};
+
+    # check device UP (or UNKNOWN for some machines)
+    if (script_run("ip link show dev $dev | egrep 'state (UP|UNKNOWN)'")) {
+        die "Network device '$dev' should be UP (or UNKNOWN)";
+    }
+    # check IPv6 addresses in use
+    die 'IPv6 address not assigned' if script_output("ip -6 addr show dev $dev") eq '';
+    # check assigned to the default zone (public)
+    if (script_output("firewall-cmd --list-interfaces") ne "$dev") {
```
Comment (sofiasyria) [inline tests/console/verify_network.pm:35]: a bit more flexible option, just for possible future enhancement of the workers, would be to check if the output includes "$dev".

File: tests/console/verify_network.pm [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+use cfg_files_utils 'validate_cfg_file';
+
+sub run {
+    select_console 'root-console';
+    my ($nm, $dev, $cfg_files) = @{get_test_suite_data()->{network}}{
+        qw(network_manager device config_files)};
+
+    # check device UP (or UNKNOWN for some machines)
+    if (script_run("ip link show dev $dev | egrep 'state (UP|UNKNOWN)'")) {
+        die "Network device '$dev' should be UP (or UNKNOWN)";
```
Comment (sofiasyria) [inline tests/console/verify_network.pm:30]: isn't there a way to know what we expect and include it in test_data?

**Review by sofiasyria — APPROVED** [review]
LGTM

File: tests/console/verify_network.pm (hunk 3 above)
Comment (jknphy) [reply tests/console/verify_network.pm:35]: in case of more interfaces the test should be modified, so only one interface is expected in the default zone for now.

File: tests/console/verify_network.pm (hunk 4 above)
Comment (jknphy) [reply tests/console/verify_network.pm:30]: For what I could check with other teams, this is seen with PowerVM, machines with bridge or ovs interfaces. I decided to relax the expectations as we could schedule this module in other scenarios. AFIK it seems acceptable 'unknown'.
